nncf/experimental/torch/nas/bootstrapNAS/search/search.py

```diff
@@ -0,0 +1,386 @@
+                evaluator.input_model_value = value
+                if value > self.search_params._ref_acc - 0.01 or value < self.search_params._ref_acc - 0.01:
+                    nncf_logger.warning(f"Accuracy obtained from evaluation {value} differs from reference accuracy {self.search_params._ref_acc}")
+                    if self.search_params._ref_acc == 100:
+                        nncf_logger.info("Adjusting reference accuracy to accuracy obtained from evaluation")
+                        self.search_params._ref_acc = value
+                    elif self.search_params._ref_acc < 100: # REMOVE. Not Needed because we want to make a distiction between both values.
+                        nncf_logger.info("Using reference accuracy.")
+                        evaluator.input_model_value = self.search_params._ref_acc
+            else:
+                if evaluator.use_model_for_evaluation:
+                    value = evaluator.evaluate_model(self._model)
+                else:
+                    value = evaluator.evaluate_with_elasticity_handler()
+                evaluator.input_model_value = value
+
+    def search_progression_to_csv(self, filename='search_progression.csv') -> NoReturn:
+        with open(f'{self._log_dir}/{filename}', 'w') as progression:
+            writer = csv.writer(progression)
+            for record in self._search_records:
+                writer.writerow(record)
+
+    def save_evaluators_state(self) -> NoReturn:
+        evaluators_state = []
+        for evaluator in self._evaluators:
+            eval_state = evaluator.get_state()
+            evaluators_state.append(eval_state)
+        torch.save(Path(self.checkpoint_save_dir, 'evaluators_state.pth'))
+
+    def load_evaluators_state(self) -> NoReturn:
+        # TODO(pablo)
+        pass
+
+    def evaluators_to_csv(self) -> NoReturn:
+        for evaluator in self._evaluators:
+            evaluator.export_cache_to_csv(self._log_dir)
+
+    def _add_default_evaluators(self, validate_fn: Callable, val_loader: DataLoader) -> NoReturn:
+        self._use_default_evaluators = True
+        self._num_obj = 2
+        self._evaluators = []
+        flop_evaluator = Evaluator("flops", get_flops_for_active_subnet, 0, self._elasticity_ctrl)
+        flop_evaluator.use_model_for_evaluation = False
+        self._evaluators.append(flop_evaluator)
+        self._evaluators.append(AccuracyEvaluator(validate_fn, val_loader))
+
+
+class SearchProblem(Problem):
+    def __init__(self, search):
+        super().__init__(n_var=search.num_vars,
+                         n_obj=search._num_obj,
+                         n_constr=search.search_params._num_constraints,
+                         xl=search.vars_lower,
+                         xu=search.vars_upper,
+                         type_var=search._type_var)
+        self._search = search
+        self._elasticity_handler = self._search._elasticity_ctrl.multi_elasticity_handler
```

Comment:
21 entries of `self._search._` means that encapsulation of `SearchAlgorithm` is broken, there are some design issues.

Try to delegate some stuff to `SearchAlgorithm` or implement getters of `SearchAlgorithm`.

Reply:
Thanks! Fixing this issue